django_celery_results/backends/database.py

```diff
@@ -115,14 +117,39 @@ def _forget(self, task_id):
     def cleanup(self):
         """Delete expired metadata."""
         self.TaskModel._default_manager.delete_expired(self.expires)
+        self.GroupModel._default_manager.delete_expired(self.expires)
+
+    def _restore_group(self, group_id):
+        """return result value for a group by id."""
+        group_result = self.GroupModel._default_manager.get_group(group_id)
+
+        if group_result:
+            res = group_result.as_dict()
+            decoded_result = self.decode_content(group_result, res["result"])
+            res["result"] = (
+                [self.app.AsyncResult(tid) for tid in decoded_result] if decoded_result else None
+            )
+            return res
+
+    def _save_group(self, group_id, group_result):
+        """Store return value of group"""
+        content_type, content_encoding, result = self.encode_content([r.id for r in group_result])
+        self.GroupModel._default_manager.store_group_result(
+            content_type, content_encoding, group_id, result
+        )
+        return group_result
+
+    def _delete_group(self, group_id):
+        try:
+            self.GroupModel._default_manager.get_group(group_id).delete()
+        except self.TaskModel.DoesNotExist:
+            pass
 
     def apply_chord(self, header_result, body, **kwargs):
         """Add a ChordCounter with the expected number of results"""
         results = [r.as_tuple() for r in header_result]
         data = json.dumps(results)
```

Comment:
Once this PR is done we could use it in storing the head group for chords instead of json encoding the `header_result` here and decoding it again in `on_chord_part_return`